### DISCUSSION TITLE
support armv8 rabbitmq image?

### DISCUSSION THREAD

**Discussion by wwlleo0730** [rabbitmq/rabbitmq-delayed-message-exchange#243] [General]
### Describe the bug

i cant enable the plugin in arm64v8/rabbitmq image

the image from here  https://hub.docker.com/r/arm64v8/rabbitmq/

and versions are all latest：

arm64v8/rabbitmq:3.12.0-management and rabbitmq_delayed_message_exchange-3.12.0.ez

### Reproduction steps

the errors：

[root@kelin13 test]# docker exec -it f6870d73f867 /bin/sh

rabbitmq-plugins enable rabbitmq_delayed_message_exchange

Enabling plugins on node rabbit@f6870d73f867:
rabbitmq_delayed_message_exchange
The following plugins have been configured:
  rabbitmq_delayed_message_exchange
  rabbitmq_management
  rabbitmq_management_agent
  rabbitmq_prometheus
  rabbitmq_web_dispatch
Applying plugin configuration to rabbit@f6870d73f867...
Stack trace: 

** (CaseClauseError) no case clause matching: {:error, {:cannot_read_enabled_plugins_file, '/etc/rabbitmq/enabled_plugins', :eacces}}
    (rabbitmqctl 3.12.0-dev) lib/rabbitmq/cli/plugins/plugins_helpers.ex:112: RabbitMQ.CLI.Plugins.Helpers.update_enabled_plugins/4
    (rabbitmqctl 3.12.0-dev) lib/rabbitmq/cli/plugins/commands/enable_command.ex:127: anonymous fn/6 in RabbitMQ.CLI.Plugins.Commands.EnableCommand.do_run/2
    (elixir 1.14.4) lib/stream.ex:1529: anonymous fn/2 in Stream.iterate/2
    (elixir 1.14.4) lib/stream.ex:1742: Stream.do_unfold/4
    (elixir 1.14.4) lib/stream.ex:1813: Enumerable.Stream.do_each/4
    (elixir 1.14.4) lib/stream.ex:1051: Stream.do_transform_inner_enum/7
    (elixir 1.14.4) lib/stream.ex:1813: Enumerable.Stream.do_each/4
    (elixir 1.14.4) lib/enum.ex:2514: Enum.reduce_while/3

{:case_clause, {:error, {:cannot_read_enabled_plugins_file, '/etc/rabbitmq/enabled_plugins', :eacces}}}

### Expected behavior

so shoud i compile it myself on arm64 server?

**Comment by mkuratczyk** [comment]
It's not a bug in the plugin but a file permission issue:
`{:error, {:cannot_read_enabled_plugins_file, '/etc/rabbitmq/enabled_plugins', :eacces}}`

This may be caused by the images you are using (that we don't maintain) or by the deployment configuration.

**Comment by wwlleo0730** [comment]
it really really really hard to build a rabbitmq plugin 

i build the plugin on my arm64 server  , looks like ok now


![1](https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/assets/6244792/a0c723df-0949-4bc4-b6f9-a4bbd2f2aa28)

![2](https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/assets/6244792/24a0c5e5-fc82-4422-a7f0-f58c06a1bbd2)

**Reply by mkuratczyk** [reply]
What kind of problems did you have building it?

**Reply by wwlleo0730** [reply]
cause i am a Java developer and have no experience with Erlang

I find it quite challenging to set up the erlang development environment for the first time

and the error messages during the compilation process of the plugin are not very clear either

and finlly, i dont know how to build .ez file, make me crazy

**Reply by mkuratczyk** [reply]
Well, everything is hard for the first time and I agree that some Erlang messages can be quite cryptic, but we can't solve all the problems. My understanding is that you managed to build it, but if no - please share what you get right now

**Reply by michaelklishin** [reply]
Fortunately about a month or so (on average) after a new RabbitMQ feature (minor) release no one really has a reason to build community plugins from source any more. There is an architecture and OS-neutral binary release that can be downloaded.

**Comment by wwlleo0730** [comment]
A few days later... I found out that the real reason is that there is no permission

 It does not need to be compiled on arm64, ， just set umask and all right 

![333](https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/assets/6244792/49d1b668-f30a-42ea-8ec4-208c52a44376)
